lib/stack_master/commands/apply.rb

```diff
@@ -87,17 +97,53 @@ def create_stack
         cf.create_stack(stack_options.merge(tags: proposed_stack.aws_tags))
       end
 
+      def upload_files
+        s3.upload_files(s3_options)
+      end
+
+      def template_method
+        return :template_body unless use_s3?
+        :template_url
+      end
+
+      def template_value
+        return proposed_stack.maybe_compressed_template_body unless use_s3?
+        s3.url(bucket: @s3_config['bucket'], prefix: @s3_config['prefix'], region: @s3_config['region'], template: @stack_definition.template)
+      end
+
+      def files_to_upload
+        return {} unless use_s3?
+        {
+          @stack_definition.template => {
+            :path => @stack_definition.template_file_path,
+            :body => proposed_stack.maybe_compressed_template_body
```

Comment:
A better name for this method `maybe_compressed_template_body` is `compiled_template_body`.